Sources/Shared/Combine/VimeoClient+Publisher.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  VimeoClient+Publisher.swift
+//  VimeoNetworking
+//
+//  Created by Rogerio de Paula Assis on 11/17/19.
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+extension VimeoClient {
+
+    /// Returns a publisher that wraps a decodable request for a given URLConvertible type.
+    /// - Parameter request: the request convertible for which to create a decodable request.
+    ///
+    /// The publisher publishes the specified decodable type when the request completes and immediately
+    /// sends a completion event. If the request fails, it terminates with an error.
+    @available(iOS 13, tvOS 13, macOS 10.15, *)
+    func publisher<D: Decodable>(for endpoint: EndpointType) -> AnyPublisher<D, Error> {
+        let passthroughSubject = PassthroughSubject<D, Error>()
+        var task: RequestToken?
+        task = self.request(endpoint) { (result: Result<D, Error>) in
+            switch result {
+            case .success(let decodable):
+                passthroughSubject.send(decodable)
+                passthroughSubject.send(completion: .finished)
```

Comment:
Is it always the case that we'd want to send `.finished` when a request returns successfully?

I know we have a heartbeat system which posts data to the server related to whether a user is watching a livestream. It seems like with this method we'd have to create a new publisher after each request, even if we don't care about the result. I guess this is sort of related to my `VIMNullResponse` comment above.